src/libraries/Microsoft.Extensions.Logging.Abstractions/src/IBufferedLogger.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Logging.Abstractions
+{
+    /// <summary>
+    /// Logging providers can implement this interface to indicate they support buffered logging.
+    /// </summary>
+    /// <remarks>
+    /// A logging provider normally exposes an <see cref="ILogger" /> interface that gets invoked by the
+    /// logging infrastructure whenever it’s time to log a piece of state.
+    ///
+    /// The logging infrastructure will type-test the <c>ILogger</c> object to determine if
+    /// it supports the <c>IBufferedLogger</c> interface also. If it does, that tells the
+    /// logging infrastructure that the logging provider supports buffering. Whenever log
+    /// buffering is enabled, buffered log records will be delivered to the logging provider
+    /// via the <c>IBufferedLogger</c> interface.
+    ///
+    /// If a logging provider does not support log buffering, then it will always be given
+    /// unbuffered log records. In other words, whether or not buffering is requested by
+    /// the user, it will not happen for those log providers.
+    /// </remarks>
```

Comment:
```suggestion
    /// <summary>
    /// Represents the ability of a logging provider to support buffered logging.
    /// </summary>
    /// <remarks>
    /// A logging provider implements the <see cref="ILogger" /> interface that gets invoked by the
    /// logging infrastructure whenever it’s time to log a piece of state.
    ///
    /// A logging provider may also optionally implement the <see cref="IBufferedLogger" /> interface.
    /// The logging infrastructure may type-test the <see cref="ILogger" /> object to determine if
    /// it supports the <see cref="IBufferedLogger" /> interface. If it does, that indicates to the
    /// logging infrastructure that the logging provider supports buffering. Whenever log
    /// buffering is enabled, buffered log records may be delivered to the logging provider
    /// in a batch via <see cref="IBufferedLogger.LogRecords" />.
    ///
    /// If a logging provider does not support log buffering, then it will always be given
    /// unbuffered log records. If a logging provider does support log buffering, whether its
    /// <see cref="ILogger" /> or <see cref="IBufferedLogger" /> implementation is used is
    /// determined by the log producer.
    /// </remarks>
```